### PR TITLE
New package: py-dask-xgboost

### PR DESCRIPTION
**Pull request by a-saitoh-fj**
New package: py-dask-xgboost

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-dask-xgboost/package.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    """Distributed training with XGBoost and Dask.distributed."""
+
+    homepage = "https://github.com/dask/dask-xgboost/"
+    pypi = "dask-xgboost/dask-xgboost-0.1.11.tar.gz"
+
+    version('0.1.11', sha256='3fbe1bf4344dc74edfbe9f928c7e3e6acc26dc57cefd8da8ae56a15469c6941c')
+
+    variant('sparse', default=False, description='Extras require')
+
+    depends_on('python', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dask-xgboost/package.py]: You can remove this line, it's implied by the base class

File: var/spack/repos/builtin/packages/py-dask-xgboost/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+class PyDaskXgboost(PythonPackage):
+    """Distributed training with XGBoost and Dask.distributed."""
+
+    homepage = "https://github.com/dask/dask-xgboost/"
+    pypi = "dask-xgboost/dask-xgboost-0.1.11.tar.gz"
+
+    version('0.1.11', sha256='3fbe1bf4344dc74edfbe9f928c7e3e6acc26dc57cefd8da8ae56a15469c6941c')
+
+    variant('sparse', default=False, description='Extras require')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dask-xgboost/package.py]: I think a better description would be something like "add sparse support" or something like that

File: var/spack/repos/builtin/packages/py-dask-xgboost/package.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    depends_on('python', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('xgboost@:0.90', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dask-xgboost/package.py]:
```suggestion
    depends_on('py-xgboost@:0.90', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
As mentioned in #21661, according to https://github.com/dask/dask-xgboost/issues/80, `py-dask-xgboost` is actually deprecated, and the functionality has been migrated to `py-xgboost+dask`. Is this package still necessary, or can it be removed?

**Comment by a-saitoh-fj** [comment]
py-dask-xgboost: Fixed what was pointed out

- Deleted the description of "depends_on('python', type=('build', 'run'))"
- Change description
- Fix dependency from xgboost to py-xgboost

Submit this recipe as it may be used in py-dask-ml.
I think it may change in the discussion of #21661, but I will submit it first.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**